src/bun.js/bindings/webcore/JSMIMEParams.cpp

```diff
@@ -0,0 +1,732 @@
+
+    // Validate name (must be a valid HTTP token)
+    int invalidNameIndex = findFirstInvalidHTTPTokenChar(nameStr);
+    if (nameStr.isEmpty() || invalidNameIndex != -1) {
+        Bun::ERR::INVALID_MIME_SYNTAX(scope, globalObject, "parameter name"_s, nameStr, invalidNameIndex);
+        RETURN_IF_EXCEPTION(scope, {});
+    }
+
+    // Validate value (must contain only valid quoted-string characters)
+    int invalidValueIndex = findFirstInvalidHTTPQuotedStringChar(valueStr);
+    if (invalidValueIndex != -1) {
+        Bun::ERR::INVALID_MIME_SYNTAX(scope, globalObject, "parameter value"_s, valueStr, invalidValueIndex);
+        RETURN_IF_EXCEPTION(scope, {});
+    }
+
+    // 2. Perform Set Operation
+    JSMap* map = thisObject->jsMap();
+    map->set(globalObject, jsString(vm, nameStr), jsString(vm, valueStr));
+    RETURN_IF_EXCEPTION(scope, encodedJSValue());
+
+    return JSValue::encode(jsUndefined());
+}
+
+JSC_DEFINE_HOST_FUNCTION(jsMIMEParamsProtoFuncDelete, (JSGlobalObject * globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto* thisObject = jsDynamicCast<JSMIMEParams*>(callFrame->thisValue());
+    if (UNLIKELY(!thisObject)) {
+        scope.throwException(globalObject, Bun::createInvalidThisError(globalObject, thisObject, "MIMEParams"));
+        RETURN_IF_EXCEPTION(scope, {});
+    }
+
+    JSValue nameValue = callFrame->argument(0);
+    String name = nameValue.toWTFString(globalObject);
+    RETURN_IF_EXCEPTION(scope, encodedJSValue());
+
+    JSMap* map = thisObject->jsMap();
+    map->remove(globalObject, jsString(vm, name));
+    RETURN_IF_EXCEPTION(scope, encodedJSValue());
+
+    return JSValue::encode(jsUndefined());
+}
+
+JSC_DEFINE_HOST_FUNCTION(jsMIMEParamsProtoFuncToString, (JSGlobalObject * globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto* thisObject = jsDynamicCast<JSMIMEParams*>(callFrame->thisValue());
+    if (UNLIKELY(!thisObject)) {
+        scope.throwException(globalObject, Bun::createInvalidThisError(globalObject, thisObject, "MIMEParams"));
+        RETURN_IF_EXCEPTION(scope, {});
+    }
+
+    JSMap* map = thisObject->jsMap();
+    StringBuilder builder;
+    bool first = true;
+
+    JSValue iteratorValue = JSMapIterator::create(globalObject, globalObject->mapIteratorStructure(), map, IterationKind::Entries);
+    RETURN_IF_EXCEPTION(scope, encodedJSValue());
+    JSMapIterator* iterator = jsDynamicCast<JSMapIterator*>(iteratorValue);
+    if (!iterator) { // Should not happen for JSMap.entries()
+        Bun::ERR::INVALID_MIME_SYNTAX(scope, globalObject, "Internal error: Expected MapIterator"_s, "toString"_s, -1);
+        RETURN_IF_EXCEPTION(scope, {});
+    }
+
+    while (true) {
+        JSValue nextValue;
+        if (!iterator->next(globalObject, nextValue)) break;
+        RETURN_IF_EXCEPTION(scope, encodedJSValue());
```

Comment:
is this exception guard getting skipped okay in the condition `break` is reached?

Reply:
next can't throw. removed